src/test/java/org/folio/rest/api/DereferencedItemStorageTest.java

```diff
@@ -0,0 +1,245 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.rest.jaxrs.model.DereferencedItem;
+import org.folio.rest.jaxrs.model.DereferencedItems;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+
+
+@RunWith(JUnitParamsRunner.class)
+public class DereferencedItemStorageTest extends TestBaseWithInventoryUtil {
+  private static final UUID smallAngryPlanetId = UUID.randomUUID();
+
+  // see also @BeforeClass TestBaseWithInventoryUtil.beforeAny()
+
+  @BeforeClass
+  public static void beforeTests() throws InterruptedException, ExecutionException, TimeoutException {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    JsonObject smallAngryPlanet = smallAngryPlanet(smallAngryPlanetId, holdingsRecordId);
+    JsonObject nod = nod(UUID.randomUUID(), holdingsRecordId);
+    JsonObject uprooted = uprooted(UUID.randomUUID(), holdingsRecordId);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    getCompleted = client.post(itemsStorageUrl(""), smallAngryPlanet, StorageTestSuite.TENANT_ID);
+    Response response = getCompleted.get(5, SECONDS);
+    assertThat(response.getStatusCode(), is(201));
+    getCompleted = client.post(itemsStorageUrl(""), nod, StorageTestSuite.TENANT_ID);
+    response = getCompleted.get(5, SECONDS);
+    assertThat(response.getStatusCode(), is(201));
+    getCompleted = client.post(itemsStorageUrl(""), uprooted, StorageTestSuite.TENANT_ID);
+    response = getCompleted.get(5, SECONDS);
+    assertThat(response.getStatusCode(), is(201));
+  }
+
+  @AfterClass
+  public static void cleanUpDatabase() {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+  }
+
+  @Test
+  public void CanGetRecordByCQLSearch() {
+    String queryString = "barcode=036000291452";
+    
+    DereferencedItems items = findByCql(queryString);
+
+    assertThat(items.getTotalRecords(), is(1));
+
+    DereferencedItem item = items.getDereferencedItems().get(0);
+
+    assertThat(item.getBarcode(), is("036000291452"));
+    assertThat(item.getId(), is(smallAngryPlanetId.toString()));
+    assertThat(item.getInstanceRecord().getTitle(), is("Long Way to a Small Angry Planet"));
+    assertThat(item.getPermanentLoanType().getName(), is("Can Circulate"));
+    assertThat(item.getMaterialType().getName(), is("journal"));
+    assertThat(item.getHoldingsRecord().getInstanceId(), is(item.getInstanceRecord().getId()));
```

Comment:
What about the other dereferenced records?

Reply:
The only one required not tested here is location-I added an assertion for that.

Reply:
What about the optional ones?

Reply:
If we're going to test for that, then we should also test for the absence of fields that should be null.  Added tests for both.

Reply:
Thanks